### PR TITLE
Enabled Athena with a dedicated S3 bucket

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

Let's have more tools to investigate issues around load balancers. AWS Athena can help us with that. [The trello card is here](https://trello.com/c/L775pbnO/243-enable-athena-engine-to-investigate-recent-load-balancers-issue).

# Test instructions | Instructions pour tester la modification

We'll log into the AWS management console and see if AWS Athena is enabled and working with queries against load balancers.

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their review | Voici une suggestion de liste de vérification comprenant des questions que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce que ça devrait être divisé en de plus petites demandes de tirage (« pull requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce changement (fichier README, etc.)?

### CODE REVIEW

**Comment by jimleroyer** [comment]
[Enable Athena engine to investigate recent load balancers' issue](https://trello.com/c/L775pbnO/243-enable-athena-engine-to-investigate-recent-load-balancers-issue)

File: aws/common/athena.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  name = "notification_athena_workgroup"
+
+  configuration {
+    enforce_workgroup_configuration    = true
+    publish_cloudwatch_metrics_enabled = true
+
+    result_configuration {
+      output_location = "s3://${aws_s3_bucket.athena.bucket}/output/"
+
+      encryption_configuration {
```
Comment (jimleroyer) [inline aws/common/athena.tf:40]: I've put some encryption configuration and went with the easiest option to use:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/athena_workgroup#encryption_configuration

Tell me if we want encryption and if that's the right option to use.

File: aws/common/athena.tf [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      days = 90
+    }
+  }
+
+  tags = {
+    CostCenter = "notification-canada-ca-${var.env}"
+  }
+
+  #tfsec:ignore:AWS002 - Ignore log of logs
+}
```
Comment (AntoineAugusti) [inline aws/common/athena.tf:67]: Can you move S3 buckets to `s3.tf`?

File: aws/common/athena.tf [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      }
+    }
+  }
+
+  lifecycle_rule {
+    id      = "expire"
+    enabled = true
+
+    expiration {
+      days = 7
```
Comment (AntoineAugusti) [inline aws/common/athena.tf:21]: 👍

File: aws/common/athena.tf [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  lifecycle_rule {
+    enabled = true
+
+    expiration {
+      days = 90
```
Comment (AntoineAugusti) [inline aws/common/athena.tf:58]: 👍

File: aws/common/athena.tf [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      }
+    }
+  }
+
+  tags = {
+    CostCenter = "notification-canada-ca-${var.env}"
+  }
+}
+
+resource "aws_athena_database" "notification_athena" {
```
Comment (AntoineAugusti) [inline aws/common/athena.tf:90]: Can you add encryption with `SSE_S3`?

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/athena_database#encryption_option

File: aws/common/athena.tf [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  tags = {
+    CostCenter = "notification-canada-ca-${var.env}"
+  }
+
+  #tfsec:ignore:AWS002 - Ignore log of logs
+}
+
+resource "aws_athena_workgroup" "notification_athena_workgroup" {
```
Comment (AntoineAugusti) [inline aws/common/athena.tf:69]: We may be able to delete the workgroup. After [reading the documentation](https://docs.aws.amazon.com/athena/latest/ug/user-created-workgroups.html), we already have a primary workgroup we cannot delete and it seems like this is enough for us.

> By default, each account has a primary workgroup and the default permissions allow all authenticated users access to this workgroup. The primary workgroup cannot be deleted.

Can you take a look at the doc and confirm you understand the same thing?

**Review by AntoineAugusti — CHANGES_REQUESTED** [review]

File: aws/common/athena.tf (hunk 4 above)
Comment (jimleroyer) [reply aws/common/athena.tf:58]: I will change to 7 to match with the previous one.

File: aws/common/athena.tf (hunk 6 above)
Comment (jimleroyer) [reply aws/common/athena.tf:69]: That makes sense.. It seems like workgroup is a feature added to have independent sandbox to manage queries, finance, quotas, permissions... we'd be good with the default in our case.

File: aws/common/athena.tf [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+resource "aws_athena_database" "notification_athena" {
+  name   = "notification_athena"
+  bucket = aws_s3_bucket.athena_bucket.bucket
+
+  encryption_configuration {
+    encryption_option = "SSE_S3"
+  }
+
+  depends_on = [aws_s3_bucket.athena_bucket]
```
Comment (AntoineAugusti) [inline aws/common/athena.tf:13]:
```suggestion
```
You should be able to leave this out as you're already referencing this resource in the bucket `field`

**Review by AntoineAugusti — APPROVED** [review]
Made a suggestion to delete an extra `depends_on`

LGTM, keep me posted if it works in staging

File: aws/common/athena.tf (hunk 7 above)
Comment (jimleroyer) [reply aws/common/athena.tf:13]: Ok got it

**Pull request merged**